src/components/Post.tsx

```diff
@@ -0,0 +1,244 @@
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const height = isShowMore
+      ? Math.min(contentHeight, maxHeroHeight)
+      : textHeight;
+    return {
+      height: withTiming(height, {
+        duration: 300,
+        easing: Easing.inOut(Easing.ease),
+      }),
+    };
+  });
+
+  const handleAnimation = () => {
+    animatedOpacity.value = withTiming(isShowMore ? 0.8 : 0, {
+      duration: 300,
+      easing: Easing.inOut(Easing.ease),
+    });
+  };
+
+  useEffect(() => {
+    animatedHeight.value = isShowMore
+      ? Math.min(contentHeight, maxHeroHeight)
+      : textHeight;
+    handleAnimation();
+  }, [isShowMore, contentHeight]);
+
+  const onTextLayout = (event: LayoutChangeEvent) => {
+    setContentHeight(event.nativeEvent.layout.height);
+  };
+
+  const onContainerPress = () => {
+    if (isShowMore) {
+      toggleShowMore();
+    }
+  };
+
+  const navigation: any = useNavigation();
+  const ImageInfo = createImageInfo(navigation, isLiked, setIsLiked);
+
+  const DescriptionView = () => {
+    if (!item.description) {
+      return null;
+    }
+    if (item.description.length > 120) {
+      return (
+        <ScrollView
+          showsVerticalScrollIndicator={false}
+          style={{flexGrow: 0}}
+          contentContainerStyle={{
+            flexGrow: 0,
+          }}>
+          <Pressable onPress={toggleShowMore}>
+            <Text
+              style={styles.description}
+              onLayout={isShowMore ? onTextLayout : undefined}>
+              {isShowMore
+                ? item.description
+                : `${item.description.slice(0, 120)}...`}
+              <Text style={styles.showText}>
+                {isShowMore ? ' Show Less' : ' Show More'}
+              </Text>
+            </Text>
+          </Pressable>
+        </ScrollView>
+      );
+    } else {
+      return (
+        <Text style={styles.description} onLayout={onTextLayout}>
+          {item.description}
+        </Text>
+      );
+    }
+  };
+
+  return (
+    <Pressable onPress={onContainerPress} style={{height, flex: 1}}>
+      <LinearGradient
+        colors={['rgba(0, 0, 0, 0.8)', 'transparent']}
+        style={styles.topGradient}
+      />
+
+      <ImageBackground
+        source={item.backgroundSource}
+        style={styles.imageBackground}
+        resizeMode="cover">
+        <Animated.View style={[styles.overlay, {opacity: animatedOpacity}]} />
+        <View style={styles.heroContainer}>
+          <View style={styles.imageAndText}>
+            <Image source={item.logoSource} style={styles.logoImage} />
+            <View style={styles.titleContainer}>
+              <Text style={styles.title}>{item.title}</Text>
+            </View>
+            <Animated.View style={[styles.descriptionContainer, animatedStyle]}>
+              <DescriptionView />
+            </Animated.View>
+          </View>
+          <View style={styles.icons}>
+            {ImageInfo.map(({id, source, text, onPress}) => (
```

Comment:
maintain the naming convention, you are using camelCase everywhere, so change this to camelCase, also prevent the use of anonymous functions, please